### PR TITLE
Help System: Implement question message pinning

### LINKED ISSUE #1020: PIn the opening message in help  channel for the session duration
Following the discussion that happened in [meta today](https://ptb.discordapp.com/channels/267624335836053506/429409067623251969/727504311638294582) I would like to suggest a new feature to the bot: make it pin the first message sent in the newly opened help session, and, of course, un-pin it when the help channel is closed.
Reasoning: some help session tend to drag out for longer time and thus to have lots of messages. This makes it very hard for anyone who's willing to to help but has not been following the discussion from the start like [here](https://ptb.discordapp.com/channels/267624335836053506/696348850981699624/727507952281452576) : having the message pinned to the channel would provide with the fast and easy way to look up the original question.

### PR DESCRIPTION
**Pull request by ks129**
Added first (claiming) message pinning and unpinning on dormant for better access when help session is long and new peoples join to helping. Resolves #1020

### CODE REVIEW

File: bot/cogs/help_channels.py [hunk 1]
```diff
@@ -548,6 +552,22 @@ async def move_to_dormant(self, channel: discord.TextChannel, caller: str) -> No
...
+            msg = discord.utils.get(self.bot.cached_messages, id=msg_id)
+            if msg is None:
+                try:
+                    msg = await channel.fetch_message(msg_id)
+                except discord.NotFound:
+                    log.debug(f"Can't unpin message {msg_id} because this is deleted.")
+
+            # When we got message, then unpin it
+            if msg:
+                await msg.unpin()
```
Comment (Senjan21) [inline bot/cogs/help_channels.py:569]: PR looks solid to me. However the lines 560-569 could be simplified, discord.py has a `HTTPClient.unpin_method` method [seen here](https://github.com/Rapptz/discord.py/blob/master/discord/http.py#L450) which is exposed under `bot.http` which we could use instead, as that will directly be making call to API without needing to try and find the message in either cache or via `fetch_message` API call and will only need to handle `HTTPException` error, which will be just raised when message wasn't found.

File: bot/cogs/help_channels.py [hunk 2]
```diff
@@ -548,6 +552,15 @@ async def move_to_dormant(self, channel: discord.TextChannel, caller: str) -> No
 
         A caller argument is provided for metrics.
         """
+        msg_id = await self.question_messages.pop(channel.id)
+
+        try:
+            await self.bot.http.unpin_message(channel.id, msg_id)
```
[the remark this answers is not in the document]
Comment (MarkKoz) [reply bot/cogs/help_channels.py:558]: Never mind, saw the other comment on this.

File: bot/cogs/help_channels.py [hunk 3]
```diff
@@ -548,6 +552,15 @@ async def move_to_dormant(self, channel: discord.TextChannel, caller: str) -> No
...
+        except discord.HTTPException:
+            log.trace(f"Message {msg_id} don't exist, can't unpin.")
```
[the remark this answers is not in the document]
Comment (Senjan21) [reply bot/cogs/help_channels.py:560]: Only other error am aware of it rising is `discord.NotFound`, which occurs only if the channel was not found, however it is very unlikely that one of our help channels would be gone, thus I don't see a reason to handle that error.

[the remark this answers is not in the document]
Comment (MarkKoz) [reply bot/cogs/help_channels.py:560]: Can we be more precise here than catching `HTTPException`? Which status code does it return when a channel isn't found? Does it have a discord-specific ID too?

[the remark this answers is not in the document]
Comment (Senjan21) [reply bot/cogs/help_channels.py:560]: >  Which status code does it return when a channel isn't found?

Judging by testing:
It returns `10003` error code 
As for `Unknown Message` `HTTPException`, the code is `10008`

**Pull request merged**